keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -944,37 +947,58 @@ class SqlDeliveryConfigRepository(
     }
   }
 
-  override fun pendingConstraintVersionsFor(deliveryConfigName: String, environmentName: String): List<String> {
+  override fun getPendingArtifactVersions(deliveryConfigName: String, environmentName: String, artifact: DeliveryArtifact): List<PublishedArtifact> {
```

Comment:
All the functions that previously returned lists of version strings now return a list of `PublishedArtifact` so they match the comparator signature in `SortingStrategy`.